### ISSUE TITLE
How can I use hattention in my own decoder-only architecture?

### ISSUE THREAD

**Issue by masc-it** [HanGuo97/log-linear-attention#1]
Hello, 
thanks for the work, it's really neat!

What are the steps to integrate hattention in a decoder-only model? 

for example, I have a CausalSelfAttention nn.Module that computes attention with a custom mask, q,k,v projections.. i mean, usual stuff with F.scaled_dot_product_attention

How can I replace the attention computation with your work? practically, with just torch.

I think that easy of reusability can play a huge role in adoption phase.
